llvm/lib/Object/ArchiveWriter.cpp

```diff
@@ -704,6 +720,8 @@ static Expected<std::vector<unsigned>> getSymbols(SymbolicFile *Obj,
       if (Map == &SymMap->Map) {
         Ret.push_back(SymNames.tell());
         SymNames << Name << '\0';
+        if (CopyImportDescriptorToECMap && isImportDescriptor(Name))
```

Comment:
You could check `SymMap->UseECMap` here instead of `CopyImportDescriptorToECMap`. When we're in `Map == &SymMap->Map` branch, we know we don't have EC object.

Reply:
Sure, that's much neater